### PR TITLE
Refactor the one inch logic to make it work everywhere for v4 and v5

### PR DESCRIPTION
**Pull request by yabirgb**
Abstract v4 code to work with v5 as v5 seems to work like v4 as for what decoding refers.

This PR moves the logic to all evm chains since v4 and v5 are deployed with the same address in all the evm chains

### CODE REVIEW

File: rotkehlchen/chain/evm/decoding/decoder.py [hunk 1]
```diff
@@ -175,6 +177,8 @@ def _add_builtin_decoders(self, rules: DecodingRules) -> None:
         normal loading?
         """
         self._add_single_decoder(class_name='Safemultisig', decoder_class=SafemultisigDecoder, rules=rules)  # noqa: E501
+        self._add_single_decoder(class_name='Oneinchv4', decoder_class=Oneinchv4Decoder, rules=rules)  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/decoder.py:180]: Hmmm we should not have it for all. Only places where 1inch has deployed. Have they deployed in all ofthe chains we support? I don't think so

File: rotkehlchen/tests/fixtures/blockchain.py [hunk 2]
```diff
@@ -102,6 +102,7 @@ def _initialize_and_yield_evm_inquirer_fixture(
     )
     if mocked_proxies is not None:
         mock_proxies(parent_stack, mocked_proxies)
+    inquirer.contracts.initialize_common_abis()
```
Comment (LefterisJP) [inline rotkehlchen/tests/fixtures/blockchain.py:105]: again with this one. I told you emove this.

File: rotkehlchen/chain/evm/decoding/decoder.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/decoder.py:180]: for v5 is deployed in all the chains we support with the same address. Checked it

for v4 is in all the chains we support but for optimism they have a different address. I've move them to every chain and adjusted on optimism

File: rotkehlchen/tests/fixtures/blockchain.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/tests/fixtures/blockchain.py:105]: I've been all the weekend fighting with this and forgot

**Review by LefterisJP — APPROVED** [review]
LGTM. Rebased, fixed conflict and fixed remnant of other PR.

**Pull request merged**